frontend/packages/kubevirt-plugin/src/components/virtualization-overview/running-vms-per-template-card/RunningVmsPerTemplateCard.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ChartDonut, ChartLabel, ChartLegend } from '@patternfly/react-charts';
+import { Card, CardBody, CardHeader, CardTitle } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { LABEL_USED_TEMPLATE_NAME } from '../../../constants';
+import { VirtualMachineModel } from '../../../models';
+import { kubevirtReferenceForModel } from '../../../models/kubevirtReferenceForModel';
+import { VMKind } from '../../../types';
+
+import './running-vms-per-template-card.scss';
+
+const getVmsPerTemplateMap = (vms, vmsLoaded) => {
+  let numVMs = 0;
+  const map = {};
+  vmsLoaded &&
+    vms.forEach((vm) => {
+      const template = vm?.metadata?.labels[LABEL_USED_TEMPLATE_NAME];
+      map[template] = map.hasOwnProperty(template) ? map[template] + 1 : 1;
+      numVMs++;
+    });
+
+  const vmCount = [];
+  const legendLabel = [];
+  const dataMap = [];
+
+  for (const templateName in map) {
```

Comment:
Nit: prefer using `for .. of` instead of `for .. in` for plain objects.

Reply:
Done!